app/src/main/java/com/example/trainingintive/viewmodels/SplashViewModel.kt

```diff
@@ -1,18 +1,17 @@
 package com.example.trainingintive.viewmodels
 
 import androidx.lifecycle.ViewModel
-import com.example.trainingintive.navigators.Navigator
 import com.example.trainingintive.navigators.SplashNavigator
 import com.example.trainingintive.util.SplashScreenEvent
 import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
 import io.reactivex.rxjava3.core.Single
 import io.reactivex.rxjava3.disposables.Disposable
 import io.reactivex.rxjava3.schedulers.Schedulers
 import java.util.concurrent.TimeUnit
+import javax.inject.Inject
 
-class SplashViewModel : ViewModel() {
+class SplashViewModel @Inject constructor(val navigator: SplashNavigator) : ViewModel() {
```

Comment:
Why is it public?

Reply:
ah, I thought dagger doesn't work with private modifier, but I see now that is true only with field injection

Reply:
That's right :)